src/Psalm/Type/Atomic/TKeyOfArray.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Psalm\Type\Atomic;
+
+use Psalm\Type\Atomic;
+
+/**
+ * Represents an offset of an array.
+ *
+ * @psalm-type ArrayLikeTemplateType = TClassConstant|TKeyedArray|TList|TArray
+ */
+class TKeyOfArray extends TArrayKey
+{
+    /** @var ArrayLikeTemplateType */
+    public $type;
+
+    /**
+     * @param ArrayLikeTemplateType $type
+     */
+    public function __construct(Atomic $type)
```

Comment:
a Single Atomic? Isn't there cases where you have to create a TKeyOfArray based on an Union of two arrays and you need to handle an Union in there?

Reply:
You're right, this is something I wanted to change later (when all tests pass), and then forgot about it!

Reply:
Done.